docs: more scope lawyering (re #40238) (#40394)
JuliaLang/julia: doc/src/manual/variables-and-scoping.md

## Local Scope

A new local scope is introduced by most code blocks (see above [table](@ref
man-scope-table) for a complete list). If such a block is syntactically nested
inside of another local scope, the scope it creates is nested inside of all the
local scopes that it appears within, which are all ultimately nested inside of
the global scope of the module in which the code is evaluated. Variables in

> outer scopes are visible from any scope they contain — meaning that they can be
> read and written in inner scopes — unless there is a local variable with the
> same name that "shadows" the outer variable of the same name. This is true even

Reviewer: ```suggestion
local scopes are visible from any scope they contain — meaning that they can be
read and written in inner scopes — unless there is an explicitly declared `local`
variable with the same name that "shadows" the outer variable. This is true even
```

I had a hard time rectifying this with my mental model... I fear we have to separate out locals vs. globals but I suspect you were trying to avoid that at this initial high-level pass.

if the outer local is declared after (in the sense of textually below) an inner
block. When we say that a variable "exists" in a given scope, this means that a
variable by that name exists in any of the scopes that the current scope is

> nested inside of, including the current one.

Reviewer: ```suggestion
nested inside of, including the current one. Global variables are slightly different;
they are similarly visible from inner scopes but can only be updated in an
inner scope if explicitly declared `global` in that scope or if that scope was
entered interactively (e.g., at the REPL or in a notebook).
```

😬
